client/admin.go

```diff
@@ -0,0 +1,196 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Marc Berhault (marc@cockroachlabs.com)
+
+package client
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/cockroachdb/cockroach/base"
+	"github.com/cockroachdb/cockroach/util"
+	"github.com/cockroachdb/cockroach/util/log"
+)
+
+const (
+	// Types of admin config to manipulate.
+
+	// Accounting expects proto.AcctConfig
+	Accounting = "acct"
+	// Permission expects proto.PermConfig
+	Permission = "perms"
+	// Quit only handles Get requests.
+	Quit = "quit"
+	// User expects proto.UserConfig
+	User = "users"
+	// Zone expects proto.ZoneConfig.
+	Zone = "zones"
+)
+
+// AdminClient issues http requests to admin endpoints.
+// TODO(marc): unify the way we handle addresses in clients.
+type AdminClient struct {
+	context    *base.Context
+	address    string
+	configType string
+}
+
+// NewAdminClient returns a new 'configType' admin client, talking to 'address'.
+func NewAdminClient(ctx *base.Context, address, configType string) *AdminClient {
+	return &AdminClient{ctx, address, configType}
+}
+
+// adminURI build a URI using the passed-in key.
+func (a *AdminClient) adminURI(key string) string {
+	uri := fmt.Sprintf("%s://%s/_admin/%s/%s", a.context.RequestScheme(),
+		a.address, a.configType, url.QueryEscape(key))
+	return uri
+}
+
+// Get issues a GET and returns the plain-text body. It cannot take a key.
+func (a *AdminClient) Get() (string, error) {
```

Comment:
OK, that seems reasonable. I'm sure it can be done without repeating nearly the entire method. Perhaps an inner method that takes the appropriate set of parameters?